### ISSUE TITLE
Top left bar animation is not synchronized with real volume

### ISSUE THREAD

**Issue by bao007fei** [iina/iina#3686]
**System and IINA version:**

- macOS Monterey version 12.3.1(m1)
- IINA 1.2.0 Build 129(m1)

**Expected behavior:**
The volume bar show exactly volume is. The delayed animation makes no sense.

**Actual behavior:**
The top left volume bar animation is delayed. Especially when use gesture.

<img width="1100" alt="Screen Shot 2022-04-07 at 11 41 37 PM" src="https://user-images.githubusercontent.com/24863266/162239889-4a45f492-9bb1-4c61-925e-6b8daea12851.png">

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**

**Comment by svobs** [comment]
I also see this problem when swiping - interestingly it only seems to be an issue when increasing the volume, and not decreasing it. The swipe lag also appears to affect the "play" slider which appears in the same place in the upper left, although seemingly with slightly less of a delay. I also found that the problem was much worse when using key repeat - see below. Unclear if there are separate issues at play here.

Tested on my 2019 16" Intel Macbook Pro.

### Swiping
Test cases (2 of 4 failing):
1. When volume is at 0%, swipe up to full volume: volume slider in on-screen controller immediately goes to correct volume, but volume slider in upper-left has an almost 1sec delay (fail)
2. When volume is at 100%, swipe down to 0 volume: both sliders go down simultaneously (good)
3. When playback is at beginning of video, swipe right to quickly seek to the end: play slider in on-screen controller quickly goes to the updated location, but play slider in upper left goes there only after a ~0.5sec delay (fail)
4. When playback is at the end of video, swipe left to beginning: both sliders go down quickly and simultaneously (good)

### Keyboard repeat
Setup
Set keyboard to the maximum repeat rate in System Preferences.
In Key Bindings, map the Up Arrow key to "Volume +2". Map the Down Arrow key to "Volume -2". Map "J" key to "Seek backward 10s (exact)". Map "L" key to "Seek forward 10 s (exact)".

Test cases (4 of 4 failing):
5. When volume is at 0%, press and hold down Up Arrow for 10 seconds: actual volume correctly increases to the maximum, but both volume sliders only update to 6% and stay that way until the key is lifted, and only then they change to the maximum (fail)
6. When volume is at 100%, press and hold Down Arrow for 10 seconds: actual volume correctly decreases to 0%, but both volume sliders only update to either 94% or 96% and stay that way until the key is lifted, and only then they change to the maximum. Although with several tests I did see it do an update to 30% once, it stayed at that value until key was lifted (fail)
7. When playback is at beginning of video, press and hold L key for 10 seconds: both play sliders begin to update, then stop (usually around the first second or so). Sometimes (inconsistently) they will update again and then stop again. The video correctly updates and reaches the end, but the sliders usually do not until the key is released.
8. When playback is at the end of video, swipe left to beginning. Similar result as (7) - sometimes a sporadic update but usually no updates until key released.

Notably, using the arrow keys to fast-forward and rewind worked correctly, but seeing as this control was set in Preferences > UI > On Screen Controller > Use left/right button for: Fast Forward/Rewind, I'm assuming it operates via a different mechanism than keys set via Key Bindings.

**Comment by lhc70000** [comment]
I think this was caused by new animations added in macOS 11 or 12. Fixed in 5bf19834ee66c7d7c902a4179ba7099cc9f25589.

**Comment by svobs** [comment]
Confirmed that this is fixed. Keyboard repeat is still a problem but will log that as a separate issue

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.
